devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+# Copyright (C) 2019 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+# Thanks to Stefano Dafarra <stefano.dafarra@iit.it> for this CMakeLists.txt
+
+# List the subdirectory (http://stackoverflow.com/questions/7787823/cmake-how-to-get-the-name-of-all-subdirectories-of-a-directory)
+macro(SUBDIRLIST result curdir)
+  file(GLOB children RELATIVE ${curdir} ${curdir}/*)
+  set(dirlist "")
+  foreach(child ${children})
+    if(IS_DIRECTORY ${curdir}/${child})
+      list(APPEND dirlist ${child})
+    endif()
+  endforeach()
+  set(${result} ${dirlist})
+endmacro()
+
+# Get list of models
+subdirlist(subdirs ${CMAKE_CURRENT_SOURCE_DIR}/robots/)
+# Install each model
+foreach (dir ${subdirs})
+  yarp_install(DIRECTORY robots/${dir} DESTINATION ${YARP_ROBOTS_INSTALL_DIR})
+endforeach ()
```

Comment:
We have already a macro for the in files https://github.com/dic-iit/bipedal-locomotion-framework/blob/67b78ab1f6942df37077c038f8c09242d2a09f46/cmake/InstallIniFiles.cmake. We can either extend it, rename it, or reuse part of the functionalities.

Reply:
Thanks for the heads up! This is fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/130/commits/2b370c9e5fef2f69d9fcb9c2a5ee55028a72ee08

Reply:
To be honest, I haven't fully understood the modifications. The lines you deleted were not needed?

Reply:
I just deleted the lines for the macro that already exists as you pointed out. This macro is used in the subsequent lines.

Reply:
Ah I see now, thanks!